test/e2e/storage/csi_mock_volume.go

```diff
@@ -1988,3 +2107,35 @@ func getVolumeLimitFromCSINode(csiNode *storagev1.CSINode, driverName string) in
 	}
 	return 0
 }
+
+// checkDeleteSnapshotSecrets checks if delete snapshot secrets annotation is applied to the VolumeSnapshotContent.
+func checkDeleteSnapshotSecrets(cs clientset.Interface, annotations interface{}) error {
+	ginkgo.By("checking if delete snapshot secrets annotation is applied to the VolumeSnapshotContent")
+
+	var (
+		annDeletionSecretName      string
+		annDeletionSecretNamespace string
+		ok                         bool
+		err                        error
+
+		// CSISnapshotterDeleteSecretNameAnnotation is the annotation key for the CSI snapshotter delete secret name in VolumeSnapshotClass.parameters
+		CSISnapshotterDeleteSecretNameAnnotation string = "snapshot.storage.kubernetes.io/deletion-secret-name"
+
+		// CSISnapshotterDeleteSecretNamespaceAnnotation is the annotation key for the CSI snapshotter delete secret namespace in VolumeSnapshotClass.parameters
+		CSISnapshotterDeleteSecretNamespaceAnnotation string = "snapshot.storage.kubernetes.io/deletion-secret-namespace"
+	)
+
+	if annDeletionSecretName, ok = annotations.(map[string]interface{})[CSISnapshotterDeleteSecretNameAnnotation].(string); !ok {
```

Comment:
Won't this cause panics if annotations does not match this type? It seems like this could be split into two checks.

Reply:
updated